src/components/REPLInput.tsx

```diff
@@ -0,0 +1,251 @@
+ */
+interface REPLInputProps {
+  history: (string | JSX.Element)[];
+  setHistory: Dispatch<SetStateAction<(string | JSX.Element)[]>>;
+  mode: string;
+  setMode: Dispatch<SetStateAction<string>>;
+  setCurrentDataset: Dispatch<SetStateAction<string[][]>>;
+  currentDataset: string[][];
+}
+
+// Mocked datasets
+const mockedDatasets = {
+  path1: [
+    ["Column1", "Column2", "Column3"],
+    ["Row1Cell1", "Row1Cell2", "Row1Cell3"],
+    ["Row2Cell1", "Row2Cell2", "Row2Cell3"],
+  ],
+  path2: [
+    ["Header1", "Header2", "Header3"],
+    ["Data1", "Data2", "Data3"],
+    ["Data4", "Data5", "Data6"],
+    ["Data4", "Data7", "Data6"],
+  ],
+};
+
+// // Function to render an HTML table from the dataset
+// function renderTable(dataset: string[][]) {
+//   return (
+//     <table>
+//       <tbody>
+//         {dataset.map((row, rowIndex) => (
+//           <tr key={rowIndex}>
+//             {row.map((cell, cellIndex) => (
+//               <td key={cellIndex}>{cell}</td>
+//             ))}
+//           </tr>
+//         ))}
+//       </tbody>
+//     </table>
+//   );
+// } // i moved this to CommandHandlers
+
+/**
+ * the REPLInput component
+ *
+ * @param param0 the set of props taken in by REPLInput
+ * @returns the component itself
+ */
+export default function REPLInput({
+  history,
+  setHistory,
+  mode,
+  setMode,
+  setCurrentDataset,
+  currentDataset,
+}: REPLInputProps) {
+  const [commandString, setCommandString] = useState("");
+
+  const context = {
+    setHistory,
+    setCurrentDataset,
+    setMode,
+    currentMode: mode,
+    mockedDatasets,
+    currentDataset,
+  };
+
+  /**
+   * A function to handle what happens after the submit button is pressed
+   *
+   * @param commandString the raw string input from the input form
+   */
+  function handleSubmit(commandString: string) {
+    const trimmedCommand = commandString.trim().toLowerCase();
+    const [command, ...args] = trimmedCommand.split(" ");
+
+    if (Object.hasOwnProperty.call(commandHandlers, command)) {
+      const handler = commandHandlers[command as keyof typeof commandHandlers];
+      const result = handler(args, context);
+
+      if (Array.isArray(result)) {
+        setHistory((prevHistory) => [...prevHistory, ...result.flat()]);
+      } else {
+        setHistory((prevHistory) => [...prevHistory, result]);
+      }
+    } else {
+      const isVerbose = mode === "verbose";
+      setHistory((prevHistory) => [
+        ...prevHistory,
+        isVerbose
+          ? `Command: ${command}\nOutput: Unknown command`
+          : `Unknown command: ${command}`,
+      ]);
+    }
+
+    setCommandString("");
+  }
+
+  return (
+    <div className="repl-input">
+      <fieldset>
+        <legend>Enter a command:</legend>
+        <ControlledInput
+          value={commandString}
+          setValue={setCommandString}
+          ariaLabel={"Command input"}
+        />
+      </fieldset>
+      <button onClick={() => handleSubmit(commandString)}>
```

Comment:
every component should maybe have its own aria-label / className? This would be helpful for a developer to be able to add specific stylings for a specific button in their css, as well as improve accessibility for people who need to know what a button is specifically used for, but can't see the screen